readinglist/resource.py

```diff
@@ -198,30 +278,30 @@ def put(self):
 
         new_record = self.request.validated
         new_record = self.process_record(new_record, old=self.record)
-        self.record = self.db.update(record_id=record_id,
+        self.record = self.db.update(record_id=self.record_id,
                                      record=new_record,
                                      **self.db_kwargs)
         return self.record
 
     @resource.view(permission='readwrite')
     @exists_or_404()
+    @unmodified_or_412()
     @validates_or_400()
+    @refresh_revision()
     def patch(self):
-        record_id = self.request.matchdict['id']
-        self.record = self.db.get(record_id=record_id, **self.db_kwargs)
-
         updated = self.merge_fields(changes=self.request.json)
 
         updated = self.process_record(updated, old=self.record)
 
-        record = self.db.update(record_id=record_id,
+        record = self.db.update(record_id=self.record_id,
                                 record=updated,
                                 **self.db_kwargs)
         return record
 
     @resource.view(permission='readwrite')
```

Comment:
I kinda feel we're starting to stack a lot of decorators here. The rule I usually apply for decorator usage is: can the function can work and be understood without the decorators. If not it means that you might have too much logic pushed in decorator - making the code hard to read. But OTHO if the decorator names are clear enough...